test/box-luatest/gh_8650_add_is_sync_option_test.lua

```diff
@@ -0,0 +1,319 @@
+local t = require('luatest')
+local cluster = require('luatest.replica_set')
+local server = require('luatest.server')
+local net = require('net.box')
+local msgpack = require('msgpack')
+
+local g = t.group('gh_8650')
+
+g.before_each(function(cg)
+    cg.cluster = cluster:new({})
+    local box_cfg = {
+        memtx_use_mvcc_engine = true,
+        replication = {
+            server.build_listen_uri('master', cg.cluster.id),
+            server.build_listen_uri('replica', cg.cluster.id)
+        },
+        replication_synchro_timeout = 0.2,
+        replication_synchro_quorum = 2
+    }
+
+    cg.master = cg.cluster:build_and_add_server({alias = 'master',
+                                                 box_cfg = box_cfg})
+    box_cfg.read_only = true
+    cg.replica = cg.cluster:build_and_add_server({alias = 'replica',
+                                                  box_cfg = box_cfg})
+    cg.cluster:start()
+end)
+
+g.after_each(function(cg)
+    cg.cluster:drop()
+end)
+
+g.test_box_begin_commit_is_sync = function(cg)
+    cg.master:exec(function()
+        box.schema.space.create('test')
+        box.space.test:create_index('pk')
+        box.ctl.promote()
+
+        box.begin()
+            box.space.test:insert{1, 'is_sync = false'}
+        box.commit()
+        box.begin({is_sync = true})
+            box.space.test:insert{2, 'is_sync = true'}
+        box.commit()
+        box.begin()
+            box.space.test:insert{3, 'is_sync = true'}
+        box.commit({is_sync = true})
+        box.begin({is_sync = true})
+            box.space.test:insert{4, 'is_sync = true'}
+        box.commit({is_sync = true})
+    end)
+
+    t.assert_error_msg_content_equals('Illegal parameters, is_sync ' ..
+                                      'can only be true',
+    function()
+        cg.master:exec(function()
+            box.begin({is_sync = false})
```

Comment:
this function always fails on the first `box.begin({is_sync = false})` without testing the other variants.

You have to call 4 different functions with all possible erroneous argument combinations.

Reply:
Fixed